public/app/plugins/datasource/prometheus/components/PromQueryField.tsx

```diff
@@ -151,6 +153,26 @@ class PromQueryField extends React.PureComponent<PromQueryFieldProps, PromQueryF
     }
   }
 
+  getHints(datasource: PrometheusDatasource, data: PanelData, query: PromQuery) {
+    let queryHints;
+    const result = isDataFrame(data.series[0]) ? data.series.map(toLegacyResponseData) : data.series;
+    if (this.props.range) {
+      const start = datasource.getPrometheusTime(this.props.range.from, false);
+      const end = datasource.getPrometheusTime(this.props.range.to, true);
+      const range = Math.ceil(end - start);
+      const interval = rangeUtil.intervalToSeconds(datasource.interval);
+      const stepInterval = rangeUtil.intervalToSeconds(query.interval ?? datasource.interval);
```

Comment:
i think `datasource.interval` should not be used here.
try something like below:
(though in general i think this is getting too complicated, also my change does not work because `datasource.templateSrv` is a private variable, so we cannot access it from the outside)

```suggestion
    const request = this.props.data?.request;
    const intervalMs = request?.intervalMs;
    if (this.props.range && request !== undefined && intervalMs !== undefined) {
      const start = datasource.getPrometheusTime(this.props.range.from, false);
      const end = datasource.getPrometheusTime(this.props.range.to, true);
      const range = Math.ceil(end - start);
      const interval = intervalMs / 1000.0;
      const stepInterval = rangeUtil.intervalToSeconds(
        datasource.templateSrv.replace(query.interval ?? request.interval, request.scopedVars)
      );
```

Reply:
Yeah i agree, it's starting to get a little bit complicated. I've tried use the other approach with the red info-box but i haven't figured out how to create the error and send it to the panel yet